### PR TITLE
Move macOS matrix to the calling workflow

### PR DESCRIPTION
**Pull request by webknjaz**
Previously, part of it was passed to the reusable workflow where
another part was being populated. Now, the entire matrix is defined
in a single place, in the calling workflow.

### CODE REVIEW

File: .github/workflows/reusable-macos.yml [hunk 1]
```diff
@@ -8,13 +8,14 @@ on:
         required: false
         type: boolean
         default: false
-      os-matrix:
-        required: false
+      os:
+        description: OS to run the job in
```
Comment (hugovk) [inline .github/workflows/reusable-macos.yml]: Just a little nit, both are fine, this a bit shorter:
```suggestion
        description: OS to run the job
```

Comment (webknjaz) [reply .github/workflows/reusable-macos.yml]: @hugovk do we really need to shorten things here, given that it's not displayed in the UI, except in autocomplete/validation? Or does your variant feel more natural?

Comment (hugovk) [reply .github/workflows/reusable-macos.yml]: I think mine is a bit more natural, but if you feel strongly we can keep yours.

Comment (webknjaz) [reply .github/workflows/reusable-macos.yml]: No, yours is fine. I just wanted to understand better. I was also considering "under" instead of "in".

**Comment by hugovk** [comment]
Thanks again!

**Comment by Mapuppy09** [comment]
Change Azure to Github , thank you!

**Pull request merged**